lib/vector-core/src/config/mod.rs

```diff
@@ -673,4 +671,18 @@ mod test {
         new_definition.assert_valid_for_event(&valid_event);
         new_definition.assert_valid_for_event(&invalid_event);
     }
+
+    #[test]
+    fn test_new_log_source_ignores_definition_with_metric_data_type() {
```

Comment:
Maybe the tests naming could be improving, i'm open for suggestions on how to name & test this behavior

Reply:
I think these tests are good 👍